File: tests/scripts/verify_website_rag.py
import asyncio
import logging
import sys
import os

# Setup Path
sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), "../..")))

from config_system import config

# Ensure we use specific config for test
config.rag.enabled = True

from zena_mode.scraper import WebsiteScraper
from zena_mode.rag_pipeline import AsyncLocalRAG

# Configure Logs
logging.basicConfig(
    level=logging.INFO,
    format="%(asctime)s [%(name)s] %(levelname)s: %(message)s",
    handlers=[logging.StreamHandler(sys.stdout)],
)
logger = logging.getLogger("VerifyRAG")


async def run_verification():
    """Run verification."""
    print("=" * 60)
    print("🕷️  WEBSITE RAG VERIFICATION START")
    print("=" * 60)

    # 1. Initialize RAG
    print("\n[1] Initializing Vector DB...")
    rag = AsyncLocalRAG()
    # [X-Ray auto-fix] print(f"✅ RAG Initialized. Collection: {rag.collection_name}")
    # 2. Scrape Website (Real Scraper!)
    target_url = "https://example.com"
    # [X-Ray auto-fix] print(f"\n[2] Scraping {target_url}...")
    scraper = WebsiteScraper(target_url)
    # Run in thread as per UI logic
    result = await asyncio.to_thread(scraper.scrape, max_pages=1)

    if not result["success"]:
        # [X-Ray auto-fix] print(f"❌ Scraping Failed: {result.get('error')}")
        return

    docs = result["documents"]
    # [X-Ray auto-fix] print(f"✅ Scraped {len(docs)} documents.")
    # [X-Ray auto-fix] print(f"   - Title: {docs[0]['title']}")
    # [X-Ray auto-fix] print(f"   - Content Preview: {docs[0]['content'][:50]}...")
    # 3. Index Content
    print("\n[3] Ingesting to Qdrant...")
    await rag.build_index_async(docs)
    print("✅ Indexing Complete")

    # 4. Prove It (Search)
    print("\n[4] Searching for content...")
    query = "What is this domain for?"
    results = await rag.search_async(query, k=3)

    # [X-Ray auto-fix] print(f"✅ Found {len(results)} matches.")
    for i, r in enumerate(results):
        # [X-Ray auto-fix] print(f"   [{i + 1}] Score: {r['score']:.3f} | Text: {r['text'][:60]}...")
        pass
    if any("example" in r["text"].lower() for r in results):
        print("\n🎉 SUCCESS: Website content retrieved from RAG!")
    else:
        print("\n⚠️ WARNING: Content mismatch.")

    print("=" * 60)


if __name__ == "__main__":
    asyncio.run(run_verification())

File: rust_output/src/verify_website_rag.rs
use anyhow::{Result, Context};
use crate::config_system::{config};
use crate::rag_pipeline::{AsyncLocalRAG};
use crate::scraper::{WebsiteScraper};
use tokio;

pub static LOGGER: std::sync::LazyLock<String /* logging::getLogger */> = std::sync::LazyLock::new(|| Default::default());

/// Run verification.
pub async fn run_verification() -> () {
    // Run verification.
    println!("{}", ("=".to_string() * 60));
    println!("{}", "🕷️  WEBSITE RAG VERIFICATION START".to_string());
    println!("{}", ("=".to_string() * 60));
    println!("{}", "\n[1] Initializing Vector DB...".to_string());
    let mut rag = AsyncLocalRAG();
    println!("✅ RAG Initialized. Collection: {}", rag.collection_name);
    let mut target_url = "https://example.com".to_string();
    println!("\n[2] Scraping {}...", target_url);
    let mut scraper = WebsiteScraper(target_url);
    let mut result = asyncio.to_thread(scraper::scrape, /* max_pages= */ 1).await;
    if !result["success".to_string()] {
        println!("❌ Scraping Failed: {}", result.get(&"error".to_string()).cloned());
        return;
    }
    let mut docs = result["documents".to_string()];
    println!("✅ Scraped {} documents.", docs.len());
    println!("   - Title: {}", docs[0]["title".to_string()]);
    println!("   - Content Preview: {}...", docs[0]["content".to_string()][..50]);
    println!("{}", "\n[3] Ingesting to Qdrant...".to_string());
    rag.build_index_async(docs).await;
    println!("{}", "✅ Indexing Complete".to_string());
    println!("{}", "\n[4] Searching for content...".to_string());
    let mut query = "What is this domain for?".to_string();
    let mut results = rag.search_async(query, /* k= */ 3).await;
    println!("✅ Found {} matches.", results.len());
    for (i, r) in results.iter().enumerate().iter() {
        println!("   [{}] Score: {:.3} | Text: {}...", (i + 1), r["score".to_string()], r["text".to_string()][..60]);
        // pass
    }
    if results.iter().map(|r| r["text".to_string()].to_lowercase().contains(&"example".to_string())).collect::<Vec<_>>().iter().any(|v| *v) {
        println!("{}", "\n🎉 SUCCESS: Website content retrieved from RAG!".to_string());
    } else {
        println!("{}", "\n⚠️ WARNING: Content mismatch.".to_string());
    }
    println!("{}", ("=".to_string() * 60));
}
